src/wasm/wasm.cpp

```diff
@@ -898,6 +902,20 @@ void Push::finalize() {
   }
 }
 
+void TupleMake::finalize() {
+  std::vector<Type> types;
```

Comment:
can use `handleUnreachableOperands()` but i guess that would be two loops then (doubtful the optimizer can fuse them). it would be shorter and perhaps clearer though?

Reply:
I would write this:

```
  type = Type::none;
  handleUnreachableOperands(this);
  if (type == Type::unreachable) {
    return;
  }

  std::vector<Type> types;
  for (auto* op : operands) {
    types.push_back(op->type);
  }
  type = Type(types);
```

I'm not sure that's actually easier to understand.

Reply:
Agreed, good point.